### PR TITLE
feat: use strings to name steps

### PR DESCRIPTION
**Pull request by douglance**
- swaps from using id numbers to strings to make urls look better and less confusing for users

### CODE REVIEW

File: src/hooks/useStep.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const useStep = () => {
 
     while (currentStep) {
       sortedSteps.push(currentStep);
-      currentStep = steps.find((step) => step.id === currentStep?.next);
+      currentStep = steps.find((step) => step.id == currentStep?.next);
```
Comment (dewanshparashar) [inline src/hooks/useStep.tsx:86]: NIT: Why are we getting rid of `===`? Types seem to be consistent for steps.

**Review by dewanshparashar — APPROVED** [review]

File: src/hooks/useStep.tsx [hunk 2]
```diff
@@ -65,12 +65,12 @@ export const useStep = () => {
     }
   };
 
-  const currentStepId = pathname ? parseInt(getLastPartOfPath(pathname)) : FIRST_STEP.id;
+  const currentStepId = pathname ? getLastPartOfPath(pathname) : FIRST_STEP.id;
```
Comment (chrstph-dvx) [inline src/hooks/useStep.tsx:68]: Can we use nextjs built-in system instead? In https://github.com/OffchainLabs/arbitrum-orbit-deployment-ui/blob/main/app/deployment/step/%5Bid%5D/page.tsx we should be able to get the current step with
```
const router = useRouter()
const { step }= router.query
```

https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes

File: src/hooks/useStep.tsx [hunk 3]
```diff
@@ -65,12 +65,12 @@ export const useStep = () => {
...
 
   const chainStepMap: Record<string, Step> =
     chainType === ChainType.Rollup ? RollupStepMap : AnyTrustStepMap;
 
-  const findStepById = (id: number): Step | undefined => {
+  const findStepById = (id: string): Step | undefined => {
```
Comment (chrstph-dvx) [inline src/hooks/useStep.tsx:73]: NIT: Maybe we should reconsider how we store those steps. Array of ids for the order and map for easy access?

File: src/hooks/useStep.tsx (hunk 2 above)
Comment (chrstph-dvx) [reply src/hooks/useStep.tsx:71]: My bad, the link was for pages router and not app router. The param is passed directly to the page component:
```
export default function StepPage({ params }: { params: { id: string } }) {
  console.log('P', params.id); // will log the current step (as a string)
```

**Review by chrstph-dvx — APPROVED** [review]

**Pull request merged**